utils/credentials/credentials.go

```diff
@@ -87,6 +84,33 @@ func LoadCertificates() ([]tls.Certificate, *x509.CertPool) {
 	return []tls.Certificate{certificate}, certPool
 }
 
+// generateFromCA generates a client certificate from the provided CA.
+func generateFromCA() ([]tls.Certificate, *x509.CertPool) {
+	caEnt, err := entity.FromFile(*ca, *key)
+	if err != nil {
+		log.Exitf("Failed to load certificate and key from file: %v", err)
+	}
+	clientEnt, err := entity.CreateSigned("client", nil, caEnt)
+	if err != nil {
+		log.Exitf("Failed to create a signed entity: %v", err)
+	}
+	caPool := x509.NewCertPool()
+	caPool.AddCert(caEnt.Certificate.Leaf)
+	return []tls.Certificate{*clientEnt.Certificate}, caPool
+}
+
+// LoadCertificates loads certificates from files or from the CA.
+func LoadCertificates() ([]tls.Certificate, *x509.CertPool) {
+	if *ca != "" && *key != "" {
```

Comment:
there needs to be a differentiation between the two keys. There is one for the CA and one for the client/target. This is important to highlight.

	ca             = flag.String("ca", "", "CA certificate file.")
	caKey             = flag.String("ca_key", "", "CA private key file.")
	cert           = flag.String("cert", "", "Certificate file.")
	key            = flag.String("key", "", "Private key file.")